### ISSUE TITLE
auro-modal: Allow inset-padding adjustment

### ISSUE THREAD

**Issue by clockeby** [AlaskaAirlines/auro-dialog#11]
## Is your feature request related to a problem? Please describe.
The base padding included within the Auro modal can make the modal larger than necessary for the content included.

## Describe the solution you'd like
Allowing the inset-padding to be adjusted to any of the allowed Auro values would help the modal display at an appropriate size for the included content. A range of equal values as well as squish & stretch values can give teams basic padding guidelines which will keep the overall appearance consistent while also allowing for a greater range of size variations to fit more use cases. The base value should take into account the minimum padding allowed for each set of breakpoints (xs & sm to md, lg, etc.)

## Describe alternatives you've considered
Keeping a minimum value established by each breakpoint and allowing for only squish/stretch values beyond that may also fix the issue though I'm not sure how different this would be from an implementation standpoint.

## Additional context
Screenshot, Auro modal with 32px inset-padding.
![image](https://user-images.githubusercontent.com/9092702/95131259-9b39e300-0712-11eb-8fe2-64823972e8b1.png)

**Comment by blackfalcon** [comment]
I am concerned about the individual customization you are looking for. It is starting to feel like we are getting away from what the dialog is intended to be and more to a random series of interactions that open into a dialog-like experience. 

I suggest one of two alternatives to this request. 

1. Is this something that can be addressed with the updates proposed in https://github.com/AlaskaAirlines/auro-popover/issues/6
1. Is this a new type of dialog interaction? Another common thread of dialog is a process dialog that is meant more towards the use of enabling features in a control panel or choosing options.

Going with what is established with the concept of a dialog, and then a modal dialog, the idea of a process dialog fits the theme and intent of the interactions.

**Comment by clockeby** [comment]
The popover has less padding, but as of the most recent update, the modal dialog inset padding for LG ad MD is 64px on all sides. I know designers love their white space but 64px seems like a lot. The base grid margin and gutter for those sizes in only 32px. It seems like using those for the modal dialog when it appears within those breakpoints would be enough.

**Comment by blackfalcon** [comment]
I don't think that this is the right place to have this conversation. Currently, there is no design leadership monitoring this thread and I am not in a position to make design decisions. 

I feel that there is a much larger discussion to be had in regards to dialog boxes and the proposed updates to the popover component. 

I highly suggest that you (@clockeby) have a discussion with @Crea-Ive and other designers to come up with a new design proposal. 

I still stand behind my initial assessment that opening this component up to a wide range of customization for inset padding and header sizes will result in a wide range of custom dialog types that will not be easily repeated and drastically different from one experience to the next. 

NOTE: I am placing a BLOCKED on this issue until there is a design resolution presented.

**Comment by blackfalcon** [comment]
I placed the BLOCKED label on this back in October. Is there a resolution to this issue? Are there any action items? 

if there are no action items to take from this issue, please close.
